src/razor/src/csharp/csharpProjectedDocument.ts

```diff
@@ -88,6 +91,49 @@ export class CSharpProjectedDocument implements IProjectedDocument {
         return false;
     }
 
+    // add resolve provisional dot if a provisional completion request was made
+    // A resolve provisional dot is the same as a provisional dot, but it remembers the
+    // last provisional dot inserted location and is used for the roslyn.resolveCompletion API
+    public addResolveProvisionalDotAt() {
```

Comment:
Nit: I think `ensureResolveProvisionalDot` would be a better name for this. It doesn't always add, and on first reading of the handler that seemed like a potential bug, but it was just the name confusing me. Also drop the `At` since it doesn't take the index, it already knows.

Reply:
That makes sense! Thank you!